cdap-unit-test/src/test/java/co/cask/cdap/partitioned/AppWithMapReduceConsumingPartitions.java

```diff
@@ -134,6 +157,26 @@ public void beforeSubmit(MapReduceContext context) throws Exception {
       PartitionedFileSet lines = batchPartitionConsumer.getConfiguredDataset(context, "lines");
       context.setInput("lines", lines);
 
+      Map<String, String> outputArgs = new HashMap<>();
+      PartitionKey partitionKey = PartitionKey.builder().addLongField("time", context.getLogicalStartTime()).build();
+      PartitionedFileSetArguments.setOutputPartitionKey(outputArgs, partitionKey);
+
+      // We know that PartitionedFileSet is an OutputFormatProvider, so we set our own instance of an
+      // OutputFormatProvider (that delegates to the PFS's implementation) as an output to the MapReduce job
+      // to test MapReduceContext#addOutput(String, OutputFormatProvider)
+      final PartitionedFileSet outputLines = context.getDataset("outputLines", outputArgs);
+      context.addOutput("outputLines", new OutputFormatProvider() {
```

Comment:
I wasn't sure if the call (without casting) was ambiguous, since the object is both a Dataset and OutputFormatProvider.

If I put the cast, intellij says that its redundant (and so someone might easily remove it in the future)?